src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -265,6 +270,26 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var transactionState = transaction.getState();
+    var calculatedDueDate = transaction.getHold().getNeedBefore();
+    var requestedDueDate = borrowerRenew.getDueDateTime();
+
+    if (calculatedDueDate >= requestedDueDate) {
+      transaction.setState(BORROWER_RENEW);
+      transactionRepository.save(transaction);
+    } else if (transactionState == ITEM_SHIPPED) {
```

Comment:
Why did you add checking `transactionState == ITEM_SHIPPED`?

Reply:
If I understand correctly, documentation says that recall request from owning site to central server should be execute only if state of the transaction is `ITEM_SHIPPED`

Reply:
Do you mean this logic should be inside recall request?

Reply:
Removed